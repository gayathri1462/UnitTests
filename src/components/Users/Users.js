import { useEffect, useState } from "react";
import axios from "axios";

function Users() {
  const [users, setUsers] = useState([]);

  // Load the data from the server
  useEffect(() => {
    let mounted = true;

    const getUsers = async () => {
      const response = await axios.get(
        "https://jsonplaceholder.typicode.com/users"
      );
      if (mounted) {
        setUsers(response.data);
      }
    };

    getUsers();

    return () => {
      mounted = false;
    };
  }, []);

  return (
    <div className="Users">
      <h1>Users:</h1>
      {users.length ? (
        <ul data-testid="user-list">
          {users.map((user) => (
            <li key={user.id} className="user" data-testid="user-item">
              <span>{user.name}</span>
            </li>
          ))}
        </ul>
      ) : (
        <div>Loading users...</div>
      )}
    </div>
  );
}

export default Users;
