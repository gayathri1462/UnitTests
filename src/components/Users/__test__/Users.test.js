import axios from "axios";
//import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import Users from "../Users";
import { BrowserRouter } from "react-router-dom";
import { act, render, screen, waitFor } from "@testing-library/react";

jest.mock("axios");

const fakeUsers = [
  {
    id: 1,
    name: "Test User 1",
    username: "testuser1",
  },
  {
    id: 2,
    name: "Test User 2",
    username: "testuser2",
  },
];

const MockUsers = () => {
  return (
    <BrowserRouter>
      <Users />
    </BrowserRouter>
  );
};

describe("Users component", () => {
  test("it renders", async () => {
    axios.get.mockResolvedValue({ data: fakeUsers });

    render(<MockUsers />);
    await act(async () => {
      expect(screen.getByText("Users:")).toBeInTheDocument();
    });
    // await waitFor(() => {
    // expect(screen.getByText("Users:")).toBeInTheDocument();
    //});
  });

  test("it renders a correct snapshot", async () => {
    axios.get.mockResolvedValue({ data: fakeUsers });
    const tree = renderer.create(<MockUsers />).toJSON();
    await act(async () => {
      expect(tree).toMatchSnapshot();
    });
  });

  test("it displays a list of users", async () => {
    axios.get.mockResolvedValue({ data: fakeUsers });

    render(<MockUsers />);

    const userList = await waitFor(() => screen.getByTestId("user-list"));

    expect(userList).toBeInTheDocument();
  });

  test("it displays a row for each user", async () => {
    axios.get.mockResolvedValue({ data: fakeUsers });
    render(<MockUsers />);

    const userList = await waitFor(() => screen.findAllByTestId("user-item"));
    expect(userList).toHaveLength(2);
  });
});
