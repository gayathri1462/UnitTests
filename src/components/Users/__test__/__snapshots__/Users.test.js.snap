// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component it renders a correct snapshot 1`] = `
<div
  className="Users"
>
  <h1>
    Users:
  </h1>
  <div>
    Loading users...
  </div>
</div>
`;
